Week2/exercise_1.js

```diff
@@ -0,0 +1,99 @@
+//C:\Users\knowl\Documents\hyf\databases\Week2\exercise_1.js
+import {createNewConnection, useDatabase} from './connection_query.js';
+
+
+const createAuthorsTable = (connection)  => {
+    return new Promise((resolve, reject) => {
+        const createAuthorsTableQuery = `
+            CREATE TABLE IF NOT EXISTS authors (
+                author_id INT AUTO_INCREMENT PRIMARY KEY,
+                author_name VARCHAR(100) NOT NULL,
+                university VARCHAR(100),
+                date_of_birth DATE,
+                h_index INT,
+                gender ENUM('Male', 'Female', 'Other')
+            );
+        `;
+        connection.query(createAuthorsTableQuery, (err, results) => {
+            if (err) {
+                console.error('Error creating authors table:', err.stack);
+                reject(err);
+                return;
+            }
+            console.log('Authors table created.');
+            resolve();
+        });
+    });
+};
+
+const addMentorColumn = (connection)  => {
+    return new Promise((resolve, reject) => {
+        const checkColumnExistsQuery = `
+            SELECT COUNT(*) AS columnExists
+            FROM information_schema.columns
+            WHERE table_name = 'authors'
+              AND column_name = 'mentor';
+        `;
+
+        connection.query(checkColumnExistsQuery, (err, results) => {
+            if (err) {
+                console.error('Error checking for mentor column:', err.stack);
+                reject(err);
+                return;
+            }
+
+            const columnExists = results[0].columnExists;
+
+            if (columnExists) {
+                console.log('Mentor column already exists. No changes made.');
+                resolve();
+            } else {
+                const addMentorColumnQuery = `
```

Comment:
Was adding foreign key column conditionally part of the assignment? Because you could have created the foreign key column while creating the table, removing all the code related to conditions. Could've been simpler